src/script/lib-build.sh

```diff
@@ -72,7 +72,7 @@ function discover_compiler() {
     local cxx_compiler=g++
```

Comment:
same comment about commit title: please use the actual filename.  Also the title should be "start search for clang versions with 17 rather than 14" ("bump up discover_compiler" doesn't describe what's changed)